dbt/include/snowflake/macros/adapters.sql

```diff
@@ -155,14 +155,41 @@
     show terse objects in {{ schema_relation }}
   {%- endset -%}
 
+  {%- set fallback_sql -%}
+     select
+      created as "created_on",
+      table_name as "name",
+      case when table_type = 'BASE TABLE' then 'table'
+           when table_type = 'VIEW' then 'view'
+           when table_type = 'MATERIALIZED VIEW' then 'materializedview'
+           when table_type = 'EXTERNAL TABLE' then 'external'
+           else table_type
+      end as "kind",
+      table_catalog as "database_name",
+      table_schema as "schema_name"
+    from  information_schema.tables
+    where table_schema ilike '{{ schema_relation.schema.lower() }}'
+      and table_catalog ilike '{{ schema_relation.database.lower() }}'
+  {%- endset -%}
+
   {%- set result = run_query(sql) -%}
   {% set maximum = 10000 %}
   {% if (result | length) >= maximum %}
     {% set msg %}
       Too many objects in schema  {{ schema_relation }}! dbt can only get
-      information about schemas with fewer than {{ maximum }} objects.
+      information about schemas with fewer than {{ maximum }} objects using
+      the "show terse objects" command.
+
+      Falling back to querying the Snowflake information schema directly.
+
+      To improve performance, we recommend confirming all tables and
+      views in {{ schema_relation}} are current.
```

Comment:
I think this needs to be `{{ schema_relation }}`? Otherwise the missing space throws an error.

Reply:
Hey @mikealfare thanks for reviewing this. I learned late yesterday that in fact we can paginate the `show terse objects` query on snowflake, so shifted direction there in working with @dbeatty10.